docs/developers/guides/token.md

```diff
@@ -0,0 +1,1320 @@
+        ),
+      "'from' has not authorized transfer",
+      error.error_code.authorization_failure
+    );
+
+    const fromBalance = this._balancesStorage.get(from)!;
+
+    System.require(
+      fromBalance.value >= value,
+      "'from' has insufficient balance"
+    );
+
+    const supply = this._supplyStorage.get()!;
+
+    const newSupply = SafeMath.sub(supply.value, value);
+
+    supply.value = newSupply;
+    fromBalance.value -= value;
+
+    this._supplyStorage.put(supply);
+    this._balancesStorage.put(from, fromBalance);
+
+    const burnEvent = new token.burn_event(from, value);
+    const impacted = [from];
+
+    System.event(
+      "koinos.contracts.token.burn_event",
+      Protobuf.encode(burnEvent, token.burn_event.encode),
+      impacted
+    );
+
+    return new token.empty_message();
+  }
+}
+
+
+```
+
+We have to add `assembly/state` directory where we define our storage. 
+
+``` sh title="aseembly/state"
+BalancesStorage.ts
+SpaceIds.ts
+SupplyStorage.ts
+```
+
+Create the above files and add this code to each:
+
+```ts linenums="1" title="assembly/state/BalancesStorage.ts"
+
+import { Storage } from '@koinos/sdk-as';
+import { token } from '../proto/token';
+import { BALANCE_SPACE_ID } from './SpaceIds';
+
+export class BalancesStorage extends Storage.Map<Uint8Array, token.balance_object> {
+  constructor(contractId: Uint8Array) {
+    super(
+      contractId, 
+      BALANCE_SPACE_ID, 
+      token.balance_object.decode, 
+      token.balance_object.encode,
+      // default balance is 0
+      () => new token.balance_object()
+    );
+  }
+}
+
+```
+
+```ts linenums="1" title="assembly/state/SpaceIds.ts"
+
+export const SUPPLY_SPACE_ID = 0;
+export const BALANCE_SPACE_ID = 1;
+
+```
+
+```ts linenums="1" title="assembly/state/SupplyStorage.ts"
+
+import { Storage } from '@koinos/sdk-as';
+import { token } from '../proto/token';
+import { SUPPLY_SPACE_ID } from './SpaceIds';
+
+export class SupplyStorage extends Storage.Obj<token.balance_object> {
+  constructor(contractId: Uint8Array) {
+    super(
+      contractId, 
+      SUPPLY_SPACE_ID, 
+      token.balance_object.decode, 
+      token.balance_object.encode,
+      // total supply is 0 by default
+      () => new token.balance_object()
+    );
+  }
+}
+
+```
+
+---
+## Customizing the token
+Let's customize the specifics of our token project by modifying `token/assembly/Token.ts`. Define the following:
+
+- `_name`: The name of your token
+
+- `_symbol`: The symbol for your token
+
+- `_decimals`: The decimal places for your token
+
+
+```ts linenums="1" title="token/assembly/Token.ts"
+- _name: string = "My Token Name";
+- _symbol: string = "MTN";
+- _decimals: u32 = 8;
+```
+
+As an example we changed the `[token name]` variable to `My Token Name` and the `symbol` variable to `MTN`.
+
+
+After making your changes, update the token name and symbol in the unit test file located at `token/assembly/__tests__/Token.spec.ts`.
```

Comment:
Are we using `Token.spec.ts` or `token.spec.ts` because these are two different files... file names are case sensitive in Unix and some MacOS systems.

Also, we're now outside the `token` directory like the previous comment.

Reply:
This is how the `koinos-sdk-as-cli create token` generates the file names, I changed where it was referred to as `token.spec.ts`